src/js/components/TaskDetailComponent.jsx

```diff
@@ -223,12 +239,92 @@ var TaskDetailComponent = React.createClass({
         </dl>
         {this.getTaskHealthComponent()}
         <hr />
-        <h3>Working Directory</h3>
-        <TaskFileListComponent task={task} />
+        {this.getTabs(task)}
       </div>
     );
   },
 
+  getTabs: function (task) {
+    var appId = this.props.appId;
+    var app = AppsStore.getCurrentApp(appId);
+
+    var activeTabId =
+      `apps/${encodeURIComponent(appId)}/${encodeURIComponent(task.id)}`;
+    var activeTab = this.context.router.getCurrentParams().tab;
+
+    var tabs = tabsTemplate.map(function (tab) {
+      var id = tab.id.replace(":appId", encodeURIComponent(appId))
+        .replace(":taskId", encodeURIComponent(task.id));
+
+      if (activeTabId == null || tab === activeTab) {
+        activeTabId = id;
+      }
+
+      return {
+        id: id,
+        text: tab.text
+      };
+    });
+
+    if (activeTab === "volumes") {
+      activeTabId += "/volumes";
+    } else if (activeTab != null) {
+      activeTabId = activeTab;
+    }
+
+    var volumes = [];
+
+    if (task.localVolumes != null) {
+      volumes = task.localVolumes
```

Comment:
This chain looks a bit repetitive compared to this I saw in AppPageComponent. Is it possible to make this generic and reusable?